mcx/sample.py

```diff
@@ -390,8 +390,7 @@ def sample_scan(
         f"Collecting {num_samples:,} samples across {num_chains:,} chains",
         refresh=False,
     )
-    print_rate = int(num_samples / 20)
-    progress_bar_scan = progress_bar_factory(progress_bar, print_rate)
+    progress_bar_scan = progress_bar_factory(progress_bar, num_samples)
```

Comment:
you need `print_rate` along with `num_samples` in the progress bar, so I just passed `num_samples` here and calculate `print_rate` in `progress_bar_factory`